### DISCUSSION TITLE
Circuit - multiple piezos

### DISCUSSION THREAD

**Discussion by bzass** [corrados/edrumulus#21] [Q&A] [answered]
Hi,

As I don't fully understand the circuit, I'd rather confirm - for attaching additional piezos should we use schema 1, schema 2 or something else? I assume that schema 1 but @corrados, could you confirm? 

What is actually the reason we are using 3.3V pin - is it to move the point of operation? In most circuits with piezos I've seen the piezo was connected between GND and Analog IN through parallel diode / pull down resistor and another resistor in series to limit the current.

Thanks!

Schema 1

![edrumulus_new2_schem](https://user-images.githubusercontent.com/12543258/146455256-bd771acf-88ce-408c-a58d-4d9605808057.png)

Schema 2

![edrumulus_new3_schem](https://user-images.githubusercontent.com/12543258/146455275-8764c6d9-0df1-4319-9bb8-21392bf00183.png)

**Comment by corrados** [comment, marked as answer]
This is the correct one:

> Schema 2
> 
> ![edrumulus_new3_schem](https://user-images.githubusercontent.com/12543258/146455275-8764c6d9-0df1-4319-9bb8-21392bf00183.png)


> What is actually the reason we are using 3.3V pin - is it to move the point of operation?

Yes, exactly.

**Reply by ignotus666** [reply]
Isn't the sleeve of Piezo Trigger 2 missing a ground connection?

**Reply by bzass** [reply]
Yes you are right - I forgot to draw it. Thanks for confirming!

**Reply by thijstriemstra** [reply]
And what about, let's say 6 piezzo's, where do I add additional ones in the circuit?

**Reply by corrados** [reply]
See the PDF in that post: https://github.com/corrados/edrumulus/discussions/21#discussioncomment-1843574

BTW: Can you please post a picture of your circuit. I'm curious.

**Reply by thijstriemstra** [reply]
> BTW: Can you please post a picture of your circuit. I'm curious.

![circuit1](https://user-images.githubusercontent.com/305679/146772138-3185dfb7-86bd-43b0-ba21-21d4b559d250.jpg)
![circuit2](https://user-images.githubusercontent.com/305679/146772154-5e766c46-ef3b-490e-ab19-8d63c47f821f.jpg)
![circuit3](https://user-images.githubusercontent.com/305679/146772166-40ba0eb6-26cc-481a-ac0b-ea765b028dd6.jpg)

**Reply by corrados** [reply]
That looks good :+1:. The way you have connected the analog circuit with the ESP32, you can simply exchange the ESP32 with the Teensy which is a good idea.

I hope in the end you will get an Edrumulus setup so that you are satisfied with the trigger results.

**Reply by thijstriemstra** [reply]
thanks for the encouraging words :)

Should I switch to the master branch and figure out what specific pads (6 max) this build will support, build the circuit and test it with hairless midi? Does the master branch also provide data I can monitor on the serial plotter?

**Reply by corrados** [reply]
The main branch is a bit risky since I do experiments there. I have created a separate branch for your prototype: https://github.com/corrados/edrumulus/tree/thijstriemstra_prototype. I prepared it from the information I got from your posts. If you need updates, simply do the changes and create a pull request for that branch.

> Does the master branch also provide data I can monitor on the serial plotter?

No, you cannot have the plotter and MIDI at the same time.

> [...] build the circuit and test it with hairless midi?

The Open E-Drums project has written a nice page for usage of Hairless MIDI: https://open-e-drums.tumblr.com/post/171304647319/using-hairless-midi

**Reply by thijstriemstra** [reply]
> I have created a separate branch for your prototype:

Nice!

I'm planning to use the following setup for an initial build:

| ID | Pad | Type | Triggering | GPIO |
| --- | --- | --- | --- | --- |
| 1 | Drum-tec Diabolo | Snare | Dual | 36, 35 |
| 2 | Roland KD-120BK | Kick | Single | 33 |
| 3 | Roland CY-5 | Hihat | Dual | 32, 26 |
| 4 | Roland FD-8 | Hihat controller | Single | 25 |
| 5 | Roland PD-5 | Tom 1 | Single | 34 |
| 6 | Roland PD-5 | Tom 2 | Single | 39 |

In the code there's this comment:

```c++
NOTE: avoid ESP32 GPIO 25/26 for piezo inputs
```

but your code uses GPIO25 for the hihat ctrl. This seems a contradiction? 

Can you I make the length of `analog_pins` 6 items, since I don't use the others?

And what about `analog_pins_rimshot`, how would mine look like?

And the section below `// my prototype setup configuration...`, how I should I configure this?

**Reply by corrados** [reply]
> I'm planning to use the following setup for an initial build:

For all Dual Triggering inputs you must define two GPIO ports.

> but your code uses GPIO25 for the hihat ctrl. This seems a contradiction?

No, it's no contradiction. The Hi-Hat control is no piezo but a poti sensor.

> Can you I make the length of analog_pins 6 items, since I don't use the others?

Sure. I already did it for your on your prototype branch:
https://github.com/corrados/edrumulus/blob/07bf51c8f5850011d6a81078d1ed080eb11e5fab/edrumulus.ino#L38

> And what about analog_pins_rimshot, how would mine look like?

See my comment above. For each Dual Trigger input you have to specify a separate GPIO port for the rim piezo. For all inputs which do not use Dual Trigger, a "-1" is inserted for analog_pins_rimshot.

> And the section below // my prototype setup configuration..., how I should I configure this?

I have already pre-configured it for you on your prototype branch based in the information I had from your forums posts. As you are using unsupported pad/trigger types, we have to play a bit with the existing types to find the one which works best. In the long run, we should add support for all of your pads in Edrumulus.

**Reply by thijstriemstra** [reply]
thanks for the feedback, I've updated the table in the previous comment. And now on to some soldering..

**Reply by corrados** [reply]
I have just updated your branch with the newest specifications: https://github.com/corrados/edrumulus/commit/d3badc829315c2b44fabfec14b5c6bf44d464b8e

**Reply by corrados** [reply]
BTW: I have configured your kick pad with PD-120 which will definitely not work as expected. We'll have to fine-tune some parameters to get it working correctly.

**Reply by thijstriemstra** [reply]
@corrados what email address can I contact you? I checked `git log` but it doesn't show a meaningful emailaddress for your commits. I'd like to send you one of my cheapo PD-5 pads (because it's christmas!) to give you something back for all your efforts so far, its been fun to work on this project. If you adjust it in git I can check it there so you don't need to paste it here). Or just check git log for my email and ping me there.

**Comment by corrados** [comment]
I have now created some [KiCad files for the complete Edrumulus prototype based on ESP32](https://github.com/corrados/edrumulus/tree/main/hardware/prototype2).

Here are the schematics in a PDF file: [edrumulus.pdf](https://github.com/corrados/edrumulus/files/7738462/edrumulus.pdf)

It would be good if someone could review the schematics. I am new to KiCad and still learning...

**Reply by corrados** [reply]
There is an updated schematics in https://github.com/corrados/edrumulus/blob/main/hardware/prototype2/edrumulus.pdf
